helperFunctions/subscribeHelpers/SalesforceAgentAssignedHelpers/SubscribeToLiveAgentHelper.ts

```diff
@@ -69,24 +72,37 @@ export class SubscribeToLiveAgent {
 						if (chatEndedReason === 'agent') {
 							await updateRoomCustomFields(this.data.room.id, { agentEndedChat: true }, this.read, this.modify);
 							this.modify.getAnalytics().sendEvent(getEventData(this.data.room.id, EventName.CHAT_CLOSED_BY_AGENT));
+							await handleEndChatCallback.handleEndChat(CloseMessages.CLOSED_BY_AGENT);
+						} else {
+							const event = getEventData(this.data.room.id, EventName.CHAT_CLOSED_BY_UNDEFINED_REASON);
+							if (event?.properties) {
+								event.properties.close_method = chatEndedReason;
+							}
+							this.modify.getAnalytics().sendEvent(event);
```

Comment:
For this event, When the chat is closed by the visitor, we receive the reason `visitor` in pullMessages.

When the visitor closes the chat we send an event from the widget. After that, the salesforce case is closed with the reason `visitor` and we receive the `ChatEnded` message as a part of pullMessages. So it will again send the duplicate event.

Same case with timeout

Reply:
So this event is redundant?

Reply:
We definitely want ChatEnded by agent event, and the widget sends chat closed by the visitor so we don't need to send it from SF. If there are any other reasons from the salesforce for chat close we need to cover those